vllm/model_executor/models/mixtral.py

```diff
@@ -318,18 +321,45 @@ def forward(
 
 
 class MixtralForCausalLM(nn.Module):
+    supports_lora = True
+    lora_target_modules = [
+        "qkv_proj",
+        "o_proj",
+        "embed_tokens",
+        "lm_head",
+    ]
+    packed_modules_mapping = {
+        "qkv_proj": [
+            "q_proj",
+            "k_proj",
+            "v_proj",
+        ],
+    }
+    embedding_modules = {
+        "embed_tokens": "input_embeddings",
+        "lm_head": "output_embeddings",
+    }
+    embedding_padding_modules = ["lm_head"]
 
     def __init__(
         self,
         config: MixtralConfig,
         linear_method: Optional[LinearMethodBase] = None,
+        lora_config: Optional[LoRAConfig] = None,
     ) -> None:
         super().__init__()
         self.config = config
         self.linear_method = linear_method
         self.model = MixtralModel(config, linear_method)
-        self.lm_head = ParallelLMHead(config.vocab_size, config.hidden_size)
-        self.sampler = Sampler(config.vocab_size)
+        self.unpadded_vocab_size = config.vocab_size
```

Comment:
Can you say why this is different from the code in `mistral.py`? There is the minor cosmetic difference of `unpadded_vocab_size` vs. `self.unpadded_vocab_size` that we should fix, but also the larger difference of the `padding_size` difference in the `ParallelLMHead` as well as the different parameters of the `Sampler`. Is there a reason why the code is not the same? :)

Reply:
should be the same, fixed